tests/h/traversal/roots_test.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import pytest
+import mock
+
+from h.models import AuthClient
+from h.services.user import UserService
+from h.traversal.roots import AnnotationRoot
+from h.traversal.roots import AuthClientRoot
+from h.traversal.roots import OrganizationRoot
+from h.traversal.roots import OrganizationLogoRoot
+from h.traversal.roots import GroupRoot
+from h.traversal.roots import UserRoot
+from h.traversal.contexts import AnnotationContext
+
+
+@pytest.mark.usefixtures('group_service', 'links_service')
+class TestAnnotationRoot(object):
+    def test_get_item_fetches_annotation(self, pyramid_request, storage):
+        factory = AnnotationRoot(pyramid_request)
+
+        factory['123']
+        storage.fetch_annotation.assert_called_once_with(pyramid_request.db, '123')
+
+    def test_get_item_returns_annotation_resource(self, pyramid_request, storage):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = mock.Mock()
+
+        resource = factory['123']
+        assert isinstance(resource, AnnotationContext)
+
+    def test_get_item_resource_has_right_annotation(self, pyramid_request, storage):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = mock.Mock()
+
+        resource = factory['123']
+        assert resource.annotation == storage.fetch_annotation.return_value
+
+    def test_get_item_raises_when_annotation_is_not_found(self, storage, pyramid_request):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = None
+
+        with pytest.raises(KeyError):
+            factory['123']
+
+    def test_get_item_has_right_group_service(self, pyramid_request, storage, group_service):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = mock.Mock()
+
+        resource = factory['123']
+        assert resource.group_service == group_service
+
+    def test_get_item_has_right_links_service(self, pyramid_request, storage, links_service):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = mock.Mock()
+
+        resource = factory['123']
+        assert resource.links_service == links_service
+
+    @pytest.fixture
+    def storage(self, patch):
+        return patch('h.traversal.roots.storage')
+
+    @pytest.fixture
+    def group_service(self, pyramid_config):
+        group_service = mock.Mock(spec_set=['find'])
+        pyramid_config.register_service(group_service, iface='h.interfaces.IGroupService')
+        return group_service
+
+    @pytest.fixture
+    def links_service(self, pyramid_config):
+        service = mock.Mock()
+        pyramid_config.register_service(service, name='links')
+        return service
+
+
+class TestAuthClientResourceFactory(object):
+    def test_get_item_returns_an_authclient(self, pyramid_request):
```

Comment:
It returns an AuthClient...model...right? Worth being explicit in the test method name perhaps? This comment also applies to the other *Root tests later in this module.